### PR TITLE
Feature/studentdashboard

### LINKED ISSUE #30: StudentForm Component
It must: 
 - [x] Have a dropdown name input
 - [x] A submit button

### LINKED ISSUE #31: Create a StudentDashboard Component
It must: 
 - [x] Have a Question
 - [x] Have possible Answers
 - [x] Render the students Name
 - [x] Render a Reading

### LINKED ISSUE #35: Create AnswerArea Component
It must:
 - [x] hold a question
 - [x] hold all the possible answers

### LINKED ISSUE #45: Create a mood form component
- [ ] Create a mood form component

### PR DESCRIPTION
**Pull request by Billwilke42**
## Pull request checklist
Please check if your PR fulfills the following requirements:
- [ ] Lint (`npm run lint`) has passed locally and any fixes were made for failures
## Pull request type
 
 
Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Testing
- [ ] Other (please describe): 
## What is the current behavior?

-  This entire feature had not been built out.
## What is the new behavior?

- A student can now take an assessment, and submit a mood. These are not connected to the database and uses mock Data to get to the next steps. Styles the components a bit using colorblind-friendly palettes. Adds to students score when the answer is correct. Fixes merge conflicts locally. Not all functionality is there including:
 - Actual thunk calls to database
 - disabling the buttons when the answer is wrong
 - the animations
 - the post to the database with the students correct answers
 -  routing

## Does this introduce a breaking change?
- [ ] Yes
- [x] No

## Other information
<<<<<<< HEAD

=======

closes #30 
closes #35 
closes #31 
closes #45
>>>>>>> 49ea6a2e4d81cd8aaf5ad430b688adb405c87982

### CODE REVIEW

File: src/StudentDashboard/MoodForm.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import { connect } from 'react-redux'
+import { lesson } from '../mockData/mockData'
+
+
+
+function MoodForm(props) {
+  const [studentMoodInput, setStudentMoodInput] = useState('')
+
+  
+  return (
```
Comment (KyleWong2510) [inline src/StudentDashboard/MoodForm.js:15]: Stoked to implement this!

File: src/StudentDashboard/QuestionArea.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import { bindActionCreators } from 'redux'
+
+function QuestionArea(props) {
+  const answers = []
+  function shuffle(theAnswers) {
+    var j, x, i;
+    for (i = theAnswers.length - 1; i > 0; i--) {
+        j = Math.floor(Math.random() * (i + 1));
+        x = theAnswers[i];
+        theAnswers[i] = theAnswers[j];
```
Comment (Melizzo) [inline src/StudentDashboard/QuestionArea.js:14]: Nice - the randomization of the answers - correct & incorrect!

File: src/StudentDashboard/StudentDashboard.scss [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  background-color: #F0E442;
+  border-radius: 5px;
+  box-shadow: 3px 2px 6px;
+  border: 5px outset #E69F00; 
+}
+
+.animation {
+  width: 40%;
+  height: 25vh;
+  padding: 10px;
```
Comment (Melizzo) [inline src/StudentDashboard/StudentDashboard.scss:45]: Super propes for the animation Bill!

File: src/StudentDashboard/MoodForm.scss [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  margin-top: 15vh;
+  height: 5vh;
+  width: 15vw;
+  background-color: #CC79A7;
+  border: none;
+  border-radius: 5px;
+  cursor: pointer;
+  border: 4px ridge #0072B2
+}
+h4 {
```
Comment (KyleWong2510) [inline src/StudentDashboard/MoodForm.scss:16]: fyi I believe that h4 style will style ALL h4s in the app, not just this component.  You could nest this or just give it a class name.

**Review by Melizzo — APPROVED** [review]

File: src/StudentDashboard/QuestionArea.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
+import React from 'react'
+import './QuestionArea.scss'
+import { incrementScore, incrementCurrentQuestion, setLessonOver } from '../actions/index';
+import { connect } from 'react-redux'
+import { bindActionCreators } from 'redux'
+
+function QuestionArea(props) {
+  const answers = []
+  function shuffle(theAnswers) {
```
Comment (KyleWong2510) [inline src/StudentDashboard/QuestionArea.js:9]: Cool shuffle function

File: src/StudentDashboard/QuestionArea.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
+import React from 'react'
+import './QuestionArea.scss'
+import { incrementScore, incrementCurrentQuestion, setLessonOver } from '../actions/index';
+import { connect } from 'react-redux'
+import { bindActionCreators } from 'redux'
+
+function QuestionArea(props) {
```
Comment (KyleWong2510) [inline src/StudentDashboard/QuestionArea.js:7]: We have been destructuring the props for readability but this is not a big deal.  We never talked about this in our standups.

File: src/StudentDashboard/StudentDashboard.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+import './StudentDashboard.scss'
+
+
+function StudentDashboard(props) {
+
+ return (
+    <section className='student-body'>
+      {props.lessonOver && <MoodForm />}
+      {!props.studentName && <StudentForm />}
+      {(props.studentName && !props.lessonOver) &&
```
Comment (KyleWong2510) [inline src/StudentDashboard/StudentDashboard.js:20]: inception!

File: src/StudentDashboard/StudentDashboard.scss [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+  flex-direction: row;
+  justify-content: space-between;
+}
+
+.app-name {
+  margin-left: 5vw;
+  font-size: 24px;
+}
+
+h4 {
```
Comment (KyleWong2510) [inline src/StudentDashboard/StudentDashboard.scss:67]: Do you see conflicts with the other h4?  Just curious.

File: src/StudentDashboard/StudentDashboard.test.js [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+    fireEvent.click(answer)
+
+    const question = getByText('What is an adjective?')
+    const answer2 = getByText('Adverb')
+    
+    expect(question).toBeInTheDocument()
+    expect(answer2).toBeInTheDocument()
+    
+  })
```
Comment (KyleWong2510) [inline src/StudentDashboard/StudentDashboard.test.js:82]: Great job with these tests.  Nice and thorough!

**Review by KyleWong2510 — APPROVED** [review]
Super impressed with you knocking this all out on your own.  Killed it!

File: src/StudentDashboard/MoodForm.scss (hunk 4 above)
Comment (Billwilke42) [reply src/StudentDashboard/MoodForm.scss:16]: Ah thanks for noticing that

File: src/StudentDashboard/QuestionArea.js (hunk 6 above)
Comment (Billwilke42) [reply src/StudentDashboard/QuestionArea.js:7]: I can be sure to do that from here on out

File: src/StudentDashboard/StudentDashboard.scss (hunk 8 above)
Comment (Billwilke42) [reply src/StudentDashboard/StudentDashboard.scss:67]: I dont think so, if you gave yours a classname it should override it

**Pull request merged**